### ISSUE TITLE
rework of pull list section

### ISSUE THREAD

**Issue by docrostov** [loathers/TourGuide#40]
the main change needed is that pulls are "one of each" now. getting the pull list to respect this new rule is probably going to be a reasonably meaty rework anyway, so as part of it, we should probably update some of the suggested pulls. a few of them are really bad at this point, and there are a good number of pulls that are not present that should (probably) be recommended. we can keep them a bit general (by ignoring some of the crazier ones, like eldritch attunement nonsense) but i still think it is a useful thing to look into and figure out.

**Comment by docrostov** [comment]
useful note from the channel: as part of this we also need to do a clover rework, specifically removing 1 or 2 mentioned clover choices that no longer exist. the most specific one of note here is the sonar in a biscuit clover, which literally does not exist anymore.

**Comment by docrostov** [comment]
new note from busta's new PR -- currently the pull list does not properly remove items that have different enchantments in 2CRS, or remove -com items from the equation for LAR where they don't work. this is a good change to track towards.

**Comment by BadHorseMonkey** [comment]
I was looking at this, planning a combination of 
1 in_ronin()
2: item not in freePulls
3: item not in _roninStoragePulls

in int pullable_amount(item it, int maximum_total_wanted)

Would you like a PR for this?

**Comment by BadHorseMonkey** [comment]
Having trouble getting set up to create the PR, but the contents are short and easy.

Pulls.ash, line 33-
```
if (in_ronin() && amount > 0 )
	{
	string[] already_pulled = split_string(get_property("_roninStoragePulls"));
	if (already_pulled contains it.to_int()) 
	    amount = 0;
	else 
	    amount = 1;
	}
```

**Comment by docrostov** [comment]
oh man i totally missed this! thanks for identifying that. i will incorporate that into the PR that will be pushed either later today or monday

**Comment by BadHorseMonkey** [comment]
Not sure if it's better.  It's working for me as expected now.

```
if (in_ronin() && amount > 0 )
	{
	string ronin_storage_pulls = get_property("_roninStoragePulls");
	string[int] already_pulled = split_string(ronin_storage_pulls);
	int requested_item = it.to_int();
	if ( already_pulled contains requested_item ) 
	    amount = 0;
	else 
	    amount = 1;
	}
```

**Comment by docrostov** [comment]
@BadHorseMonkey apologies that this took so long; needed to track down a test account that had enough stuff for me to verify that the full rework worked. but, it did! this has been resolved as of version 2.2.0, pr #184 -- please let me know in a new issue if anything is amiss in the new pull list section!
